x/emissions/keeper/actor_utils/worker.go

```diff
@@ -241,66 +200,22 @@ func insertForecastsFromTopForecasters(
 		/// Now do filters on each forecaster
 		// Ensure that we only have one forecast per forecaster. If not, we just take the first one
 		if _, ok := forecastsByForecaster[forecast.Forecaster]; !ok {
-			// Check if the forecaster is registered
-			isForecasterRegistered, err := k.IsWorkerRegisteredInTopic(ctx, topicId, forecast.Forecaster)
-			if err != nil {
-				ctx.Logger().Warn("Error checking forecaster registration: ", topicId, ", nonce: ", nonce, "for forecaster: ", forecast.Forecaster)
-				continue
-			}
-			if !isForecasterRegistered {
-				ctx.Logger().Warn("Forecaster not registered, topic: ", topicId, ", nonce: ", nonce, "for forecaster: ", forecast.Forecaster)
-				continue
-			}
-
-			// Examine forecast elements to verify that they're for inferers in the current set.
-			// We assume that set of inferers has been verified above.
-			// We keep what we can, ignoring the forecaster and their contribution (forecast) entirely
-			// if they're left with no valid forecast elements.
-			acceptedForecastElements := make([]*types.ForecastElement, 0)
-			for _, el := range forecast.ForecastElements {
-				if _, ok := acceptedInferersOfBatch[el.Inferer]; ok {
-					acceptedForecastElements = append(acceptedForecastElements, el)
-				}
-			}
-
-			// Discard if empty
-			if len(acceptedForecastElements) == 0 {
-				continue
-			}
-
-			/// Filtering done now, now write what we must for inclusion
-
-			// Get the latest score for each forecaster => only take top few by score descending
-			latestScore, err := k.GetLatestForecasterScore(ctx, topicId, forecast.Forecaster)
-			if err != nil {
-				continue
-			}
-			latestForecasterScores[forecast.Forecaster] = latestScore
+			latestForecaster = append(latestForecaster, forecast)
```

Comment:
i have another comment about this, but this easily should happen when the bundle is being committed, not when it's being closed. this adds extra, unnecessary storage load if we allow duplicate entries prior to nonce close